code-challenges.js

```diff
@@ -41,35 +61,105 @@ var vowelTester1 = "learn"
+var operations = {
+    "+": function (num1, num2) {
+        return num1 + num2
+    },
+    "-": function (num1, num2) {
+        return num1 - num2
+    },
+    "*": function (num1, num2) {
+        return num1 * num2
+    },
+    "/": function (num1, num2) {
+        return num1 / num2
+    }
+}
+//create function that takes in two num arguments plus mathematical operations
+const calculator = (number1, operator, number2 ) => {
+    // If no numbers entered return error message
+    if(number1 === null || number2 === null) {
+        return "Enter yo digits first!"
+    } 
+    //if input is not a number return error messsage
+    if(isNaN(number1) || isNaN(number2)) {
+        return "I can only calculate numeros"
+    }
+    //if divided by zero return can't divide by 0!
+    if(number2 === 0 && operator === "/"){
+        return "Can't divide by 0!"
+    } 
+    //if can perform the given operation do so and return the result
+    return operations[operator](number1, number2) 
+}
 // Uncomment and use the following console logs to test your function
 // console.log(calculator(3, "*", 9))
-// Expected output: 27
-
+// // Expected output: 27
 // console.log(calculator(16, "+", 3))
-// Expected output: 19
-
+// // Expected output: 19
 // console.log(calculator(89, "/", 0))
-// Expected output: "Can't divide by 0!"
-
-
+// // Expected output: "Can't divide by 0!"
+// console.log(calculator("a", "-", "b"))
+// Expected output: "I can only calculate numeros"
 
 // --------------------6) Create a function that takes in the following variable and returns only the websites that end in ".io".
 
 var websites = ["codepen.io", "codecademy.com", "coursera.org", "codepen.io", "udemy.com", "pluralsight.com", "udacity.com", "sitepoint.com"]
 // Expected output: "codepen.io" "codepen.io"
-
-
-
+//Declare a function that takes in array 
+const ioWebsites = (arr) => {
+    //return the ".io" values by filtering the values in the array
+    //only return the inputs that have ".io"
+    //then rejoin array into strings with a space between
+    //ensure the input is lowercase
+    //return the ".io" urls
+   return arr.filter(value => value.slice(-3) === ".io").join(" ").toLowerCase()
+   
+}
+// console.log(ioWebsites(websites))
 
 // --------------------6) STRETCH: Write a function that takes a number as an argument and uses a WHILE loop to count up to that number from 0.
+
+//Declare a function that takes in a num
+const counter = (num) => {
+    //ensure input is a number if not return error message
+    if (isNaN(num)) {
+        return "Numbers only please"
+    }
+    //Initialize variable to start at zero
+    let initialCount = 0
+    //While loop to iterate number +1
+    while(initialCount < num + 1) {
+        //Console.log count
+        console.log(initialCount)
+        //Increment count by one
+        initialCount++
+    }
+    
+}
+ counter(15)
+// //expect 0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15
+ counter(7)
```

Comment:
So unfortunately, this code would not be usable in production, because technically it is not returning anything. It logs information to the console, but that would never be seen outside of the terminal or the browser console. This is why we do not use console log for anything other than bug testing, or checking our function call.

One of the missing parts you are looking for is a return to your overall function. You would also need a few other things, but that is a good place to start.